pallets/asset-index/src/lib.rs

```diff
@@ -182,6 +196,56 @@ pub mod pallet {
             Ok(().into())
         }
 
+        #[pallet::weight(10_000)] // TODO: Set weights
+        /// Callable by an admin to remove assets from the index
+        /// and maybe transfer some IndexToken
+        ///
+        /// Caller balance and recipient are updated to allocate
+        /// the correct amount of the IndexToken
+        pub fn remove_asset(
+            origin: OriginFor<T>,
+            asset_id: T::AssetId,
+            units: T::Balance,
+            recipient: Option<MultiLocation>,
+            recipient_account: Option<T::AccountId>,
+        ) -> DispatchResultWithPostInfo {
+            T::AdminOrigin::ensure_origin(origin.clone())?;
+            let caller = ensure_signed(origin)?;
+            <Self as AssetRecorder<T::AssetId, T::Balance>>::remove_asset(
+                &asset_id, &units, recipient,
+            )?;
+
+            // Get current value from PriceFeed
+            let value = Self::calculate_pint_equivalent(asset_id.clone(), units)?;
+
+            // Updates IndexToken
+            if Self::is_liquid_asset(&asset_id) {
+                // Transfer to the recipient
+                if let Some(account) = &recipient_account {
+                    T::IndexToken::transfer(
+                        &caller,
+                        account,
+                        value,
+                        ExistenceRequirement::KeepAlive,
+                    )?;
+                } else {
+                    return Err(<Error<T>>::NoRecipient.into());
+                }
+            } else {
+                T::IndexToken::slash(&caller, value.clone());
+                T::IndexToken::burn(value);
```

Comment:
This is not ideal, because there is no guarantee that we have the account has enough balance, I think `burn` -> `settle` would be more idiomatic.